zgrab2_schemas/zgrab2/ipp.py

```diff
@@ -147,15 +147,52 @@
     "request": http_request_full
 })
 
+# TODO: Determine whether value-tag types with same underlying form should have a different name in this mapping
+ipp_attribute_value = SubRecord({
+    "raw": Binary(),
+    "integer": Signed32BitInteger(),
+    "boolean": Boolean(),
+    "enum": String(),
+    # TODO: Determine appropriate type for octetString w/o specified format
+    "octetString": Binary(),
+    "dateTime": DateTime(),
+    # TODO: Determine appropriate type for resolution
+    "resolution": Binary(),
+    # TODO: Determine appropriate type for range of Integers (probably {min, max} pair)
+    "rangeOfInteger": Binary(),
+    # TODO: Determine appropriate type for beginning of attribute collection
+    "begCollection": Binary(),
+    "textWithLanguage": String(),
+    "nameWithLanguage": String(),
+    # TODO: Determine appropriate type for end of attribute collection
+    "endCollection": Binary(),
+    "textWithoutLanguage": String(),
+    "nameWithoutLanguage": String(),
+    "keyword": String(),
+    "uri": String(),
+    "uriScheme": String(),
+    "charset": String(),
+    "naturalLanguage": String(),
+    "mimeMediaType": String(),
+    "memberAttrName": String(),
+})
+
+ipp_attribute = SubRecord({
+    "name": String(),
+    "values": ListOf(ipp_attribute_value),
+    "tag": Binary(),
```

Comment:
Is `tag` always an 8-bit unsigned integer? If so, the proper type would be `Unsigned8BitInteger()`